addons/mail/static/src/models/message_sender/message_sender.js

```diff
@@ -0,0 +1,272 @@
+            });
+            if (!this.thread.composer.isLog) {
+                postData.context = {
+                    mail_post_autofollow: true,
+                };
+            }
+            const messageId = await this.async(() =>
+                this.env.models['mail.thread'].performRpcMessagePost({
+                    postData,
+                    threadId: this.thread.id,
+                    threadModel: this.thread.model,
+                })
+            );
+            const [messageData] = await this.async(() => this.env.services.rpc({
+                model: 'mail.message',
+                method: 'message_format',
+                args: [[messageId]],
+            }, { shadow: true }));
+            this.env.models['mail.message'].insert(Object.assign(
+                {},
+                this.env.models['mail.message'].convertData(messageData),
+                {
+                    originThread: [['insert', {
+                        id: this.thread.id,
+                        model: this.thread.model,
+                    }]],
+                })
+            );
+            this.thread.loadNewMessages();
+        }
+
+        /**
+         * Function that process the message stored to be sent.
+         */
+        async _processMessagesToBeSent() {
+            if (this.isSendingMessages) {
+                return;
+            }
+            this.update({
+                isSendingMessages: true,
+            });
+            const messages = this.pendingMessagesToBeSent.filter((message) => !message.hasSendError);
+            const message = messages[0];
+            try {
+                if (message) {
+                    let postData = {
+                        attachment_ids: message.attachments.map(attachement => attachement.id),
+                        body: message.body,
+                        message_type: 'comment',
+                        partner_ids: message.partner_ids,
+                        subject: message.subject,
+                    };
+                    if (this.thread.model === 'mail.channel') {
+                        await this._postMailChannel(postData);
+                    } else {
+                        await this._postMailThread(postData);
+                    }
+                    message.delete();
+                }
+                if (!messages.length) {
+                    this.thread.refreshFollowers();
+                    this.thread.fetchAndUpdateSuggestedRecipients();
+                }
+            } catch (error) {
+                message.update({ hasSendError: true });
+                this.update({
+                    pendingMessagesToBeSent: unlink(message),
+                    MessagesNotDelivered: link(message),
+                });
+            } finally {
+                this.update({
+                    isSendingMessages: false,
+                });
+                if (messages.length > 0) {
```

Comment:
How does that condition work? Isn't the message that was just send still part of this list?

Reply:
You are right. It doesn't create bug since it's well guarded, but it trigger a `_processMessagesToBeSent` that is not needed

Reply:
Actually, since the last change, I could simply use the `this.messagesPendingToBeSent` directly. Message in error are now stored there anymore.